conans/client/tools/win.py

```diff
@@ -316,7 +314,15 @@ def vcvars_dict(settings, arch=None, compiler_version=None, force=False, filter_
             continue
         try:
             name_var, value = line.split("=", 1)
-            if not only_diff or os.environ.get(name_var) != value:
+            old_value = os.environ.get(name_var)
+            if only_diff:
+                if old_value != value:
+                    if old_value is not None and old_value in value:
```

Comment:
No need to check if ``old_value`` is in ``value``, just do the replace directly

Reply:
you have to check if old value is in new value in order to append `%<VAR_NAME>%` at the end or not